### PR TITLE
feat: adding conditional visibility for preference item

### PR DESCRIPTION
**Pull request by axel7083**
I taught at first that https://github.com/containers/podman-desktop/pull/3251/files would add a `when` condition that would allow to set some condition on other field, but apparently not. Maybe it planed in the future and therefore I could probably adapt my code. @benoitf what do you think ?

### What does this PR do?

It allows to set condition on the hidden property used in `package.json`. For example, following this PR I made adding an __advanced__ feature (https://github.com/containers/podman-desktop/pull/3508), we though it would be interesting to hide in behind something.

### Screenshot/screencast of this PR

| Switch off | Switch On |
| --- | --- |
|  image | image |

The `hidden` property can now be either a boolean or a `HiddenConfiguration`, here is an example inside a `package.json`

```json
"hidden": {
    "key": "kind.cluster.creation.switchControlPlaneImage",
    "operator": "NotEqual",
    "values": ["true"]
},
```

It is inspired by [kubernetes scheduling operators](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#operators).

> In this example only one condition is possible, but this is a very small change to add more.

### Testing

- [ ] Add unit test (todo)

### CODE REVIEW

File: extensions/kind/package.json [hunk 1]
```diff
@@ -48,11 +48,24 @@
...
           "scope": "KubernetesProviderConnectionFactory",
-          "placeholder": "Leave empty for using latest.",
+          "hidden": {
+            "key": "kind.cluster.creation.switchControlPlaneImage",
+            "operator": "NotEqual",
+            "values": [
+              "true"
+            ]
+          },
+          "placeholder": "kindest/node:v1.27.3@sha256:...",
```
Comment (benoitf) [inline extensions/kind/package.json:68]: hello @axel7083 and thanks for your pull request

I would rather not introduce a new conditional system.

we already have a context-key implementation that allow to use "natural language" like `!isLinux() && myProperty`

full list here

the when clause is used to display the element, not to hide it
https://code.visualstudio.com/api/references/when-clause-contexts#conditional-operators

Comment (axel7083) [reply extensions/kind/package.json:68]: So would you prefer to adapt the same mechanism used for the when inside the hide ? Or abandon the idea of conditional field ?

Comment (benoitf) [reply extensions/kind/package.json:68]: hi @axel7083 

I would say the `when` clause is already supporting the conditional. If the when condition is false the property is not displayed so this is why I'm a bit lost when you're saying we should abandon the idea of conditional field as for me 'when' is a conditional clause (that can reference one , multiple properties and multiple conditions)

Comment (axel7083) [reply extensions/kind/package.json:68]: > hi @axel7083
> 
> I would say the `when` clause is already supporting the conditional. If the when condition is false the property is not displayed so this is why I'm a bit lost when you're saying we should abandon the idea of conditional field as for me 'when' is a conditional clause (that can reference one , multiple properties and multiple conditions)

Thanks for your reply ! Yeah sorry I wasn't expressing myself properly, I meant abandoning/closing this PR ! 

I am looking to know if the when is meant for supporting other context such as the current form it is in ? Currently the context does seems to have much element in it, that what is losing me about it 

Also in the current implementation the function filtering on the when condition is inside the onMount therefore is not called when the form have value updated.

Moreover, I am a bit surprised of how the form is actually updating the field, the method `setRecordValue` defined here :

https://github.com/containers/podman-desktop/blob/7991d543e13d6d3171fa6a9c95b867a9776d2ac3/packages/renderer/src/lib/preferences/PreferencesRenderingItemFormat.svelte#L20

is not called for all types, which is a bit confusing, some field are updated through it, other does not seems to be (?).

Comment (benoitf) [reply extensions/kind/package.json:68]: so yes maybe there are enhancements to do to update it live (and not only at creation/first display time) but in the when clause you can already express your need

then, I would focus on having the when clause used in the Wizard tacking care of tracking changes of the expressions/contexts to hide/display items live

Comment (axel7083) [reply extensions/kind/package.json:68]: Okey thanks for the direction, I will take a look !

**Comment by axel7083** [comment]
Abandon in favor of https://github.com/containers/podman-desktop/pull/3723

**Pull request closed without merging**